src/lib/db/event-store.ts

```diff
@@ -66,13 +55,51 @@ class EventStore extends EventEmitter {
                 .returning(EVENT_COLUMNS);
             const savedEvents = savedRows.map(this.rowToEvent);
             process.nextTick(() =>
-                savedEvents.forEach(e => this.emit(e.type, e)),
+                savedEvents.forEach((e) => this.emit(e.type, e)),
             );
         } catch (e) {
             this.logger.warn('Failed to store events');
         }
     }
 
+    async delete(key: number): Promise<void> {
+        await this.db(TABLE).where({ id: key }).del();
+    }
+
+    async deleteAll(): Promise<void> {
+        await this.db(TABLE).del();
+    }
+
+    destroy(): void {}
+
+    async exists(key: number): Promise<boolean> {
+        const result = await this.db.raw(
+            `SELECT EXISTS (SELECT 1 FROM ${TABLE} WHERE id = ?) AS present`,
+            [key],
+        );
+        const { present } = result.rows[0];
+        return present;
+    }
+
+    async get(key: number): Promise<IEvent> {
+        const row = await this.db(TABLE).where({ id: key }).first();
+        return this.rowToEvent(row);
+    }
+
+    async getAll(): Promise<IEvent[]> {
+        try {
+            const rows = await this.db
+                .select(EVENT_COLUMNS)
+                .from(TABLE)
+                .limit(100)
```

Comment:
I am a bit unsure about this limit. Maybe it should be a queryArg which defaults to 100?

Reply:
Though; not relevant to this PR.

Reply:
But good question. we should probably add it as a possible query arg